soc/arm/nxp_kinetis/kv5x/linker.ld

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Linker command/script file
+ *
+ * This is the linker script for both standard images and XIP images.
+ */
+
+#include <autoconf.h>
+
+/*
+ * KV5x Flash configuration fields
+ * These are 16 bytes, which must be loaded to flash at offset 0x400, and
+ * include default protection and security settings.
+ * They are loaded at reset to various Flash Memory module (FTFE) registers.
+ */
+
+/*
+ * No need to account for this when running a RAM-only image since that
+ * security feature resides in ROM. There is also no need to account for this
+ * when building an image to be chainloaded by MCUboot, since linking fails
+ * otherwise due to the large vector table on KV5x.
+ */
+#if defined(CONFIG_XIP) && !defined(CONFIG_BOOTLOADER_MCUBOOT)
```

Comment:
Good catch. I think we can apply this logic to other kinetis platforms like k64f and save some flash.
cc: @d3zd3z